File: 283-move-zeroes/283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l = 0
        for r in range(len(nums)):
            if nums[r]:
                nums[r], nums[l], l = nums[l], nums[r], l + 1
        return
    
        l, r = 0, 0
        while r < len(nums):
            if nums[r]:
                nums[l], nums[r], l = nums[r], nums[l], l + 1
            r += 1

File: 283-move-zeroes/283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int beg = 0, end = 0;
        
        while (beg < n || end < n) {
            if (end == n) {
                nums[beg++] = 0;
                continue;
            }
            if (nums[end] != 0)
                nums[beg++] = nums[end];
            end++;
        }
    }
};